dir.c

```diff
@@ -1992,7 +1992,12 @@ rb_glob_error(const char *path, VALUE a, const void *enc, int error)
     struct glob_error_args args;
     VALUE (*errfunc)(VALUE) = glob_func_error;
 
-    if (error == EACCES) {
+#ifdef __wasi__
+    if (error == EACCES || error == ENOTCAPABLE)
+#else
+    if (error == EACCES)
+#endif
+    {
```

Comment:
Is this `#ifdef __wasi__` necessary?  Maybe we can always consider `ENOTCAPABLE` being a synonym of `EACCESS` here.

Reply:
Agree. Applied nobu's suggestion to guard by ifdef